float64 pos1
float64 pos2
float64 pos3
---
bool ret
